modules/utils/include/hephaestus/utils/unique_function.h

```diff
@@ -0,0 +1,240 @@
+//=================================================================================================
+// Copyright (C) 2023-2025 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <concepts>
+#include <cstddef>
+#include <functional>
+#include <type_traits>
+
+namespace heph {
+/// The `UniqueFunction` is very similar to `std::function` with the
+/// only difference that it is move only.
+template <typename Sig>
```

Comment:
I guess in c++23 we can replace this by std::move_only_function?

@filippobrizzi when are we finally able to migrate hephaestus to c++23 keeping it at c++20 starts to suck pretty bad

Reply:
@sithhell if std::move_only_function is a dropin replacement for this, add a comment to remove this when we update

Reply:
It could be a drop in replacement indeed. Not sure when `std::move_only_function` will land in libc++ though.